common/utils/utility-types.ts

```diff
@@ -13,6 +13,35 @@ export type OptionalToUndefined<T> = {
   [P in keyof T & keyof never]: T[P];
 };
 
+/**
+ * This allows us to take types such as:
+ * { a: string, b : string | undefined }
+ * and converts it into { a: string, b?: string | undefined }
```

Comment:
How does `{ a: string, b?: string | undefined }` differ from `{ a: string, b?: string }`

Reply:
It's how you implement the type, similar to what `Partial` does, basically saying when you have to specify a property, even if it is undefinable:
```TS
type A = { a: undefined | string }
const a = { a: undefined } // valid
const aBoo = {  } // invalid

type B = { a?: string }
const b = { a: undefined } // valid
const bYay = {  } // valid
```

Reply:
I've updated the comment to explain this.